pkg/vault/test/foundry/fuzz/FuzzHarness.sol

```diff
@@ -0,0 +1,518 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IVaultMock } from "@balancer-labs/v3-interfaces/contracts/test/IVaultMock.sol";
+import { Rounding } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+import { StablePool } from "@balancer-labs/v3-pool-stable/contracts/StablePool.sol";
+import { WeightedPool } from "@balancer-labs/v3-pool-weighted/contracts/WeightedPool.sol";
+
+import { VaultMockDeployer } from "../utils/VaultMockDeployer.sol";
+import { BasePoolMath } from "../../../contracts/BasePoolMath.sol";
+
+contract FuzzHarness is Test {
+    using FixedPoint for uint256;
+
+    event Debug(string, uint256);
+
+    // If this flag is false, assertion tests will never fail! But optimizations will work.
+    // Changing to false will disable optimizations but allow the assertion tests to run
```

Comment:
Changing to TRUE will disable optimizations but allow the assertion tests to run